### PR TITLE
Topic/convert color to grayscale test

### PR DESCRIPTION
**Pull request by nathanblair**
Reopen of #869 for @nathanblair work done. Includes a rebase on commits submitted by @imikejackson 

Uses template Parallel wrapper class for keeping algorithm decision cleaner. Files should be clang-formatted; please drop a line if you find something that isn't properly clang-formatted.

### CODE REVIEW

**Review by mmarineBlueQuartz — COMMENTED** [review]
It generally looks good, but formatting-wise, check for curly brackets for every if statement, even 1-liners, and always initialize with an equal sign unless the variable is a vector, array, or other collection.  I did notice that last one a fair number of times for `wrongParameters` and `wrongValues` in the unit test, but it was used beyond just those.

**Review by joeykleingers — APPROVED** [review]
Assuming that the test passes and tests all the features of the filter, I think this looks good.

File: Source/Plugins/Processing/Test/ConvertColorToGrayScaleTest.cpp [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+    UCharArrayType::Pointer testArray{};
+    for(const auto& eachAAName : am->getAttributeArrayNames())
+    {
+      if(eachAAName.contains(m_outputArrayPrefix))
+        testArray = am->getAttributeArrayAs<UCharArrayType>(eachAAName);
+    }
+
+    std::vector<uint8_t> colorArray{algorithmMap[algoMapIndex]};
+
+    int wrongValues{0};
```
Comment (imikejackson) [inline Source/Plugins/Processing/Test/ConvertColorToGrayScaleTest.cpp]: should be int wrongValues = 0;

File: Source/Plugins/Processing/Test/ConvertColorToGrayScaleTest.cpp [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+    tbb::parallel_for(tbb::blocked_range<size_t>(0, colorArray.size()), [&](const tbb::blocked_range<size_t>& r) {
+      for(size_t index = r.begin(); index < r.end(); ++index)
+      {
+        uint8_t testValue{testArray->getValue(index)};
```
Comment (imikejackson) [inline Source/Plugins/Processing/Test/ConvertColorToGrayScaleTest.cpp]: uint8_t testValue = testArray->getValue(index);

Please use this style for initializing simple scalar values.

**Comment by imikejackson** [comment]
All the tests pass on my macOS box. I would like Matthew's comments to be implemented. I threw in a couple of comments in the areas that Matthew is referencing.

**Comment by nathanblair** [comment]
I'll redo the initialization so that its consistent with the rest of the code base; uniform initialization is a habit for me so apologies for when it pops up. I think @mmarineBlueQuartz also mentioned in a previous PR that some of the methods were effectively empty (or perhaps that was #871?) and should be removed. Either way I'll take a look and make sure for this one as well. Always appreciate the feedback and this should be resubmitted with fixes by EOD!

**Comment by imikejackson** [comment]
Is this PR ready for merging?

**Comment by nathanblair** [comment]
Huh, I thought I had made a comment stating this was indeed ready for merging...

From my account it is. Its been clang-formatted, initialization of scalar and other trivial types uses the "=" operator. And I've verified that all of the 'for' and 'if' blocks are correctly nested inside an execution block (e.g. inside '{' and '}'). If there's anything just let me know!

**Pull request merged**